src/solver/conv_asm_1x1u.cpp

```diff
@@ -241,77 +244,210 @@ bool PerformanceConfigConvAsm1x1U::operator==(const PerformanceConfigConvAsm1x1U
         && use_spare_set == other.use_spare_set; // clang-format on
 }
 
-bool PerformanceConfigConvAsm1x1U::IsValidValue() const
+bool PerformanceConfigConvAsm1x1U::IsValidValueImpl(const int sequence_length) const
```

Comment:
This needs to be int because a negative number is passed to this to indicate special state

Reply:
I suspect this needs to be kept at `int` since the end of sequence token is -1 which needs to pass the validity test. That is why the first conditional below checks for positive value.

Reply:
Maybe I missed something, but there are no places in the code where this argument takes negative values. But what is more important here is that the name of the parameter can be misleading. Perhaps the name is good for the model, but not for this function. This can be changed later.